src/battle_script_commands.c

```diff
@@ -1223,6 +1224,19 @@ bool32 ProteanTryChangeType(u32 battler, u32 ability, u32 move, u32 moveType)
     return FALSE;
 }
 
+bool32 ShouldTeraShellDistortTypeMatchups(u32 move, u32 battlerDef)
+{
+    if (!(gBattleStruct->distortedTypeMatchups & gBitTable[battlerDef]) &&
+        GetBattlerAbility(battlerDef) == ABILITY_TERA_SHELL &&
+        gBattleMons[battlerDef].species == SPECIES_TERAPAGOS_TERASTAL &&
+        !IS_MOVE_STATUS(move) &&
+        !(gMoveResultFlags & MOVE_RESULT_NO_EFFECT) &&
+        !(gBattleMons[battlerDef].hp < gBattleMons[battlerDef].maxHP))
```

Comment:
`gBattleMons[battlerDef].hp == gBattleMons[battlerDef].maxHP`

Any reason you didn't use equality comparison here?

Also please move the `&&` operator onto a new lines in this statement as well.

Reply:
I followed Pokemon Showdown's implementation, which used !(hp < maxHP). Don't really want to change it

Reply:
Please change it since the equality comparison is much more readable. We don't have to copy bad code from showdown.

Reply:
Alright done